### PR TITLE
Add initial docker images

### PR DESCRIPTION
**Pull request by arturo-seijas**

### CODE REVIEW

**Comment by mthaddon** [comment]
I would imagine the config files are going to be written to by the charm rather than something we want to bake into the image. Let me know if you disagree about that.

Also, could we move the top level `etc` directory into a `files` directory, as if we end up with other things beside `etc` files the structure will make more sense?

A few comments inline as well.

**Comment by arturo-seijas** [comment]
> 

We'll be changing the environmental variables to adjust the config. The files that are being packed right now are not expected to change

**Comment by mthaddon** [comment]
Do we not think some of the settings in the `uwsgi.ini` might need to change at some point, e.g. `max-requests`?

**Comment by arturo-seijas** [comment]
> Do we not think some of the settings in the `uwsgi.ini` might need to change at some point, e.g. `max-requests`?

Not necessarily. But we can reassess in a later stage of development

**Review by mthaddon — COMMENTED** [review]
Some comments inline

File: etc/indico/uwsgi.ini [hunk 1]
```diff
@@ -0,0 +1,37 @@
+[uwsgi]
+;uid = indico
+;gid = indico
+;umask = 027
+;pidfile = /run/uwsgi/uwsgi.pid
+;stats = /opt/indico/web/uwsgi-stats.sock
+
+processes = 4
+enable-threads = true
+http-socket = 0.0.0.0:8080
```
[the remark this answers is not in the document]
Comment (arturo-seijas) [reply etc/indico/uwsgi.ini]: Yes, it should. Fixed

File: etc/indico/indico.conf [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+CUSTOMIZATION_DEBUG = os.environ.get('INDICO_CUSTOMIZATION_DEBUG') or False
+LOGO_URL = os.environ.get('INDICO_LOGO_URL') or ''
+
+AUTH_PROVIDERS = ast.literal_eval(os.environ.get('INDICO_AUTH_PROVIDERS', 'None')) or {}
+IDENTITY_PROVIDERS = ast.literal_eval(os.environ.get('INDICO_IDENTITY_PROVIDERS', 'None')) or {}
+LOCAL_IDENTITIES = os.environ.get('INDICO_LOCAL_IDENTITIES') or False
+
+ENABLE_ROOMBOOKING = os.environ.get('INDICO_ENABLE_ROOMBOOKING') or False
+
+PLUGINS = tuple(x for x in os.environ.get('INDICO_EXTRA_PLUGINS','').split(',') if x.strip())
```
[the remark this answers is not in the document]
Comment (arturo-seijas) [reply etc/indico/indico.conf]: Fixed

File: etc/indico/indico.conf [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+SMTP_SERVER           = (os.environ.get('SMTP_SERVER') or 'localhost', os.environ.get('SMTP_PORT') or 8025)
+SMTP_LOGIN            = os.environ.get('SMTP_LOGIN') or ''
+SMTP_PASSWORD         = os.environ.get('SMTP_PASSWORD') or ''
+SMTP_USE_TLS          = os.environ.get('SMTP_USE_TLS') or False
+
+SUPPORT_EMAIL = os.environ.get('INDICO_SUPPORT_EMAIL') or "root@localhost"
+PUBLIC_SUPPORT_EMAIL = os.environ.get('INDICO_PUBLIC_SUPPORT_EMAIL') or "root@localhost"
+NO_REPLY_EMAIL = os.environ.get('INDICO_NO_REPLY_EMAIL') or "noreply-root@localhost"
+
+LOGGING_CONFIG_FILE = "/opt/indico/etc/logging.yaml"
```
[the remark this answers is not in the document]
Comment (arturo-seijas) [reply etc/indico/indico.conf]: Changed

File: etc/indico/indico.conf [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+USE_PROXY = os.environ.get('USE_PROXY') or False
+
+LOG_DIR          = "/opt/indico/log"
+TEMP_DIR         = "/opt/indico/tmp"
+CACHE_DIR        = "/opt/indico/cache"
+
+CACHE_BACKEND = 'redis'
+REDIS_CACHE_URL = os.environ['REDIS_CACHE_URL']
+
+SMTP_SERVER           = (os.environ.get('SMTP_SERVER') or 'localhost', os.environ.get('SMTP_PORT') or 8025)
```
[the remark this answers is not in the document]
Comment (arturo-seijas) [reply etc/indico/indico.conf]: Fixed

File: etc/indico/indico.conf [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+if os.environ.get('SERVICE_PORT'):
+    BASE_URL += ':' + os.environ.get('SERVICE_PORT')
+
+ROUTE_OLD_URLS = os.environ.get('INDICO_ROUTE_OLD_URLS') or False
+USE_PROXY = os.environ.get('USE_PROXY') or False
+
+LOG_DIR          = "/opt/indico/log"
+TEMP_DIR         = "/opt/indico/tmp"
+CACHE_DIR        = "/opt/indico/cache"
```
[the remark this answers is not in the document]
Comment (arturo-seijas) [reply etc/indico/indico.conf]: Changed

File: etc/indico/uwsgi.ini [hunk 6]
```diff
@@ -0,0 +1,37 @@
+[uwsgi]
+;uid = indico
+;gid = indico
+;umask = 027
+;pidfile = /run/uwsgi/uwsgi.pid
+;stats = /opt/indico/web/uwsgi-stats.sock
```
[the remark this answers is not in the document]
Comment (arturo-seijas) [reply etc/indico/uwsgi.ini]: In fact, those can stay uncommented

**Comment by mthaddon** [comment]
LGTM, thanks. Feel free to merge when ready.

**Pull request merged**